using_images/db_images/mongodb.adoc

```diff
@@ -134,3 +134,91 @@ The MongoDB image can be run with mounted volumes to enable persistent storage f
+behave slighly different than having a regular standalone MongoDB database.
+
+The entrypoint will first discover all running member pods within the MongoDB
+cluster. For that, you have to use a 'headless' service named 'mongodb'. To define
+a headless service, you have to set the `portalIP` field value in the service
+definition to `None`. Then you can use a DNS query to get a list of the pod IP
+addresses that represents the current endpoints for this service.
+For example, you can use the `dig` command: `dig mongodb A +short +search`.
+
+When the entrypoint has the IP addresses of all running MongoDB members, you can
+create a MongoDB replication set configuration where you list all member IP
+addresses and then initiate the replication set by `rs.initiate(config)`.  Then
+you have to wait until the MongoDB elect the PRIMARY member of the cluster.  You
+can create a simple MongoDB script that will run a loop and poll the MongoDB
+server till the PRIMARY member appear on the list. Since you started the
+initiation on the 'service pod', this pod will be elected.
+
+Once the PRIMARY member has been elected, you can start creating MongoDB users and
+databases. The 'service pod' is running MongoDB without the `--auth` argument,
+so you can bootstrap the PRIMARY member without providing any authentication. 
+
+When the user accounts and databases are created and the data are replicated to
+other members, it is time for the 'service pod' to give up its PRIMARY role and
+shut down. You have to define the `restartPolicy` field in the 'service pod' to
+`Never` to prevent the 'service pod' from restarting when the container exits.
+Right after the 'service pod' goes down, other members will start a new election
+and the new PRIMARY member is being elected from the running members.
+
+The clients can then start using the MongoDB by sending the queries to the
+`mongodb` service. As this service is a 'headless' service, you don't have to
+provide the IP address. You can use `mongodb:27017` for the client
+connections. The service will then send the query to one of the members in the
+replication set.
+
+To increase the number of members in the cluster, use this command:
+
+```
+$ oc scale rc mongodb-1 --replicas=4
+```
+
+This will tell the replication controller to create a new MongoDB member pod.
+When a new member is created, the member entrypoint will first attempt do
+discover other running members in the cluster. Then it will choose one and add
+itself to the list of members. Once the replication configuration is update, the
+other members will replicate the data to a new pod and start a new election.
+
+The full source of replication example is available here: 
+https://github.com/openshift/mongodb/tree/master/2.4/examples/replica
+
+IMPORTANT: The template provided in the example above does not use persistent
+storage. When you loose all members of the replication set, your data will be
```

Comment:
I made it sound insecure intentionally as this whole template is **very** insecure if you care about the data ;-)